### ISSUE TITLE
Paragraphs tend to run together -- sometimes?

### ISSUE THREAD

**Issue by bkjproductions** [elementor/elementor#9000]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x] The issue still exists against the latest stable version of Elementor.

## Description
The text editor in the Elementor sidebar, as seen in the Tabs Element or even just a basic Text Element, is flakey. Paragraphs can run together, and it is possible to save your edited text and the paragraphs run together into one big text block. Sometimes you can see it happen when you are editing text; while in the Elementor Panel on the left, you have two returns in a row that indicates a paragraph, and on the preview panel on the right, the text will run together as if there are no paragraphs.
 

## Steps to reproduce
Toggle between Visual and Text mode when editing any text-editor-based element, such as the Tab element. At some point, the text will start running together with no returns. If you are not careful, it is possible to save the page when it is in this condition, even though you have two returns for a paragraph.  If you close the editor and come back to it, you can repair the problem by toggling or resizing the window and re-save the page. However, this is a big problem if you are just trying to do a simple text edit.

## Isolating the problem

- [x ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ x] This bug happens with a default WordPress theme active.
- [x ] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Source distribution v5.7.25
	PHP Version: 7.2.19
	PHP Max Input Vars: 1000
	PHP Max Post Size: 65M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.2.2
	Site URL: https://new.tshdlegal.com/site
	Home URL: https://new.tshdlegal.com
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: TSHD New 0.0.1
	Version: 2.1.2
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.2.0
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36

== Active Plugins ==
	Admin Columns Pro
		Version: 4.7.2
		Author: AdminColumns.com

	Advanced Custom Fields PRO
		Version: 5.8.4
		Author: Elliot Condon

	Better Search Replace
		Version: 1.3.3
		Author: Delicious Brains

	BKJ Functions
		Version: 1.2.7
		Author: Various

	Classic Editor
		Version: 1.5
		Author: WordPress Contributors

	Contact Form 7
		Version: 5.1.4
		Author: Takayuki Miyoshi

	Duplicate Page
		Version: 3.9
		Author: mndpsingh287

	Duplicate Theme
		Version: 1000.1.6bkj.01
		Author: Muller Digital Inc.

	Elementor
		Version: 2.6.8
		Author: Elementor.com

	Elementor Pro
		Version: 2.6.5
		Author: Elementor.com

	Enable Media Replace
		Version: 3.3.5
		Author: ShortPixel

	Gatherposts MT Shortcode
		Version: 2.0.7
		Author: BKJ Productions, LLC

	GitHub Updater
		Version: 8.8.2
		Author: Andy Fragen

	iThemes Security Pro
		Version: 5.9.3
		Author: iThemes

	ManageWP - Worker
		Version: 4.9.1
		Author: GoDaddy

	Open Graph
		Version: 1.9.0
		Author: Will Norris

	Remove Double Space
		Version: 0.3
		Author: Josh Eaton

	Simple CSS
		Version: 1.1
		Author: Tom Usborne

	User Switching
		Version: 1.5.2
		Author: John Blackbourn & contributors

	White Label CMS
		Version: 2.1.1
		Author: www.videousermanuals.com

	WP Client Reference
		Version: 0.42
		Author: Steve Grunwell

	Yoast SEO
		Version: 1004.9
		Author: Team Yoast

== Must-Use Plugins ==
	ManageWP - Worker Loader
		Version: 
		Author: GoDaddy

== Log ==
	: 
Log: showing 20 of 352019-06-20 14:49:23 [info] Elementor Pro/Upgrades - _v_2_5_4_form Start  
2019-06-20 14:49:23 [info] Elementor Pro/Upgrades - _v_2_5_4_form Finished 
2019-06-20 14:49:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.4.8',
  'to' => '2.5.9',
)]
2019-06-20 14:49:23 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.4.8',
  'to' => '2.5.9',
)]
2019-07-19 17:19:44 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.16',
  'to' => '2.6.5',
)]
2019-07-19 17:20:25 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.9',
  'to' => '2.5.14',
)]
2019-07-19 17:20:26 [info] elementor::elementor_updater Started 
2019-07-19 17:20:26 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Start  
2019-07-19 17:20:26 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Finished 
2019-07-19 17:20:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.16',
  'to' => '2.6.5',
)]
2019-07-30 22:08:22 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.14',
  'to' => '2.6.2',
)]
2019-07-30 22:08:22 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.6.5',
  'to' => '2.6.7',
)]
2019-07-30 22:08:22 [info] elementor::elementor_updater Started 
2019-07-30 22:08:22 [info] Elementor/Upgrades - _v_2_6_6_fa4_migration_button Start  
2019-07-30 22:08:23 [info] Elementor/Upgrades - _v_2_6_6_fa4_migration_button Finished 
2019-07-30 22:08:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.6.5',
  'to' => '2.6.7',
)]
2019-08-07 23:47:56 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.6.7',
  'to' => '2.6.8',
)]
2019-08-19 20:21:10 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.6.2',
  'to' => '2.6.3',
)]
2019-08-21 17:07:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.6.3',
  'to' => '2.6.4',
)]
2019-08-28 13:33:20 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.6.4',
  'to' => '2.6.5',
)]

JS: showing 4 of 4JS: 2019-06-21 02:56:41 [error X 1][https://new.tshdlegal.com/site/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.16:2:75476] setting getter-only property \"type\" 
JS: 2019-08-14 17:35:02 [error X 1][https://new.tshdlegal.com/site/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.6.8:32:116476] Cannot read property \'html\' of undefined 
JS: 2019-08-28 10:05:28 [error X 23][https://new.tshdlegal.com/site/wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=2.6.5:2:56801] Cannot read property \'get\' of undefined 
JS: 2019-09-04 11:55:50 [error X 1][https://new.tshdlegal.com/site/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view1094\") has already been destroyed and cannot be used. 

PHP: showing 4 of 4PHP: 2019-08-12 16:03:31 [notice X 15][/home/bhbkj2/new.tshdlegal.com/site/wp-content/plugins/elementor-pro/modules/query-control/module.php::360] Undefined index: q [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-08-13 19:09:32 [notice X 10][/home/bhbkj2/new.tshdlegal.com/site/wp-content/plugins/elementor/includes/elements/column.php::993] Undefined index: _column_size [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-08-14 18:28:53 [notice X 1][/home/bhbkj2/new.tshdlegal.com/site/wp-content/plugins/elementor-pro/modules/posts/widgets/posts.php::31] Undefined index: posts_post_type [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-08-14 20:38:07 [notice X 29][/home/bhbkj2/new.tshdlegal.com/site/wp-content/plugins/elementor-pro/modules/theme-builder/widgets/post-content.php::153] Trying to get property 'ID' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

				

</details>
![problem with the wrapping](https://user-images.githubusercontent.com/4551281/64358077-c021cb00-cfd3-11e9-8a34-80cd5632ed6c.JPG)
![bug with elementor and text fields](https://user-images.githubusercontent.com/4551281/64358078-c0ba6180-cfd3-11e9-88af-43b690d4cbca.JPG)

**Comment by bkjproductions** [comment]
![problem with the wrapping](https://user-images.githubusercontent.com/4551281/64358149-e7789800-cfd3-11e9-8a88-1a2ef06bac36.JPG)
![bug with elementor and text fields](https://user-images.githubusercontent.com/4551281/64358150-e7789800-cfd3-11e9-8659-107006911c1b.JPG)
Not sure if screenshots made it; here they are again.

**Comment by wonderlemon** [comment]
I'm experiencing this same issue. I hadn't established that it was when switching between Visual and Text editor though, so thanks for pointing that out. 
Looking forward to a fix from Elementor as this has proven to be pretty frustrating.

**Comment by bkjproductions** [comment]
Say, does anyone know how to get this ticket actually "assigned" to someone at Elementor? I just noticed that several months later it is still "unassigned".  (I'm not a regular Github user, so I have no idea; other times I've put in a ticket and it's assigned by some thoughtful person who knows how to do that...)

**Comment by bkjproductions** [comment]
@danielkatz can can you take a look at this? I have had to explain to clients that this is a bug and they are complaining. :)

**Comment by Champdor** [comment]
I would extend the scope! Text widget consistently strips most of HTML markup like `<p>`, `<span>`, `&nbsp;`. It is very annoying because I won't use separate widgets for every span, every paragraph. I think it is not the intended working. Editing only in HTML tab is no use because the next time you open the widget it is showing the Visual tab and breaks the design. Please deal with this huge bug, Elementor!

**Comment by bkjproductions** [comment]
Another month gone by and this is still not fixed? It is awkward to explain to a client, "This is a known bug."

**Comment by RonR-WebDesign** [comment]
Same issue and get really wonky with setting columns.  It won't break the paragraphs...  bummer.

**Comment by Honey-Dripper** [comment]
Elementor is not ready for prime time. Why would anyone want all there lines and spaces jammed together? Would anyone buy a magazine or a book in such an eye-straining format???? This is a joke.

**Comment by bkjproductions** [comment]
@Honey-Dripper : Well, to be fair, I see similar issues with other pagebuilders, and those don't give me the ease of use that I find in Elementor. And you can "fix" the issue by being careful when editing. ;)

It is quite vexing that a couple more months have gone by. Please spread the word and perhaps if enough people complain, Elementor will do something about it. Thanks!

**Comment by AbrahamWalker** [comment]
You're not going to like this solution but the easiest way I've found to get around this issue is to add a class to the paragraph tag.

WordPress does not strip out p tags with classes and it looks like Elementor has followed that same blueprint. 

it sucks but it is a solution.

**Comment by ShanniiWrites** [comment]
Looks like there's no solution yet! The best solution I've found is to make a completely different element and then add it in where I need via shortcode.

**Comment by bkjproductions** [comment]
@AbrahamWalker  Thanks for the tip. To clarify, you're adding a class of something, to every Paragraph you put in?
 `<p class="something">`
I guess that is workable!

**Comment by AbrahamWalker** [comment]
@bkjproductions yes, unfortunately. I made a post about this issue in the Facebook Group for Elementor users. Some users do not experience this issue. Other people have already come to the same conclusion I have.

While the process is labor intensive, it get's the job done.

**Comment by marketingspectrumNL** [comment]
Yes, same issue.
Deleting the line and manually press enter again via the visual tab fixes it most of the times. Always fighting with it.

The text editor needs some work anyway, adding <span> and <global var> tags when working within the page instead of the text editor in the widget plane.
Which end up in the source code of your page!

Told them, showed them with screenshots, but they have asked me 3 times to give a screenrecording. Happens at random. Tried it - didn’t happen. Tried it again. Didn’t happen. Gave up. Happens.

Don't know what they are up to currently anyway. With such a business, team and userbase I'd say more is possible.

Is it too much marketing and focussing on new pro users? Or is something brewing?

Latest newsletter; 
Mega menu? Here's a third party plugin! What the...

**Comment by PermaPain** [comment]
I'm having this issue too.

**Comment by mikenco** [comment]
Still an annoying issue in January 2021.

**Comment by nsepe** [comment]
Same issue. Please fix! Thank you.

**Comment by bkjproductions** [comment]
Just thought I would come in and beat this dead horse again. In later versions of Elementor Pro (3.1.0) it doesn't happen as much, but it happens enough to be awkward when showing a new Elementor user how to get around the problem.

**Comment by nsepe** [comment]
I agree. It seems I've been able to make edits without this happening.

**Comment by michaelearley1** [comment]
I am having this issue and have been for years. It's the worst part about Elementor.
Why don't we get any response from Elementor?

**Comment by i2g-lweiss** [comment]
Same issue here.  First Elementor project in a long time and I was asking myself "Why don't I use Elementor more?".  I'm starting to remember why now :(

**Comment by germinadorade** [comment]
I am having the same problem with text spans, it is very unstable.

**Comment by Xavvi** [comment]
Same problem here on multiple sites.

**Comment by d19dotca** [comment]
It's very intermittent for me, but I too have this experience from time to time.

**Comment by pradeepdotco** [comment]
Seriously annoyed with this problem now!

**Comment by d19dotca** [comment]
@DorShahar - curious why this is tagged as an enhancement and not a bug.

**Comment by raabmar** [comment]
This is really terribly annoying. Can it also be connected to the used browser?

**Comment by bkjproductions** [comment]
@raabmar, it happens regardless of browser, I have found.  
> 
> This is really terribly annoying. Can it also be connected to the used browser?

**Comment by raabmar** [comment]
I read somewhere it's connected to the editor, i.e. Elementor still uses the old built in TinyMCE editor and not Gutenberg. Installing https://en-au.wordpress.org/plugins/tinymce-advanced/ should help, as there are settings particularly for that issue. However, I did not have time yet to analyze and test it.

> @raabmar, it happens regardless of browser, I have found.

**Comment by pradeepdotco** [comment]
> @DorShahar - curious why this is tagged as an enhancement and not a bug.

Exactly! This should be a very basic thing that should be the top priority for a 'page builder'.

**Comment by rbdanny** [comment]
September 2021. Same issue as you're all finding and still I can find so resolution.
Thank you to @bkjproductions  for taking the time to state the problem more succinctly that I could have and everyone else for the comments. Good luck all.

**Comment by BluePerfectOne** [comment]
Same issue. Please fix! Thank you.

**Comment by Honey-Dripper** [comment]
This issue has been going on since 2019. Pitiful.

**Comment by ReyStub** [comment]
Thank you @raabmar,
As you mentioned, I have installed extension "Advanced Editor Tools" (previously TinyMCE Advanced), then in its settings activated "Preserve Paragraph Tags in Classic Block and Classic Editor" (this is English translation from my French environment) and it works perfectly.
Not only it solves the problem, but I get TinyMCE full features. Great ! with all the tags I am used to (P BR &nbsp marks etc. and creating tables.)
Tested on Chrome, Edge and FireFox.
French extension installed : https://fr.wordpress.org/plugins/tinymce-advanced/

**Comment by Honey-Dripper** [comment]
Why did not elementor announce this fix??? Too little and way too late. Wordpress is a joke. It begs hackers to come in and take over your site and every other site on your server. One bug and open door after another. NO THANK YOU.

**Comment by deepou** [comment]
> Thank you @raabmar, As you mentioned, I have installed extension "Advanced Editor Tools" (previously TinyMCE Advanced), then in its settings activated "Preserve Paragraph Tags in Classic Block and Classic Editor" (this is English translation from my French environment) and it works perfectly. Not only it solves the problem, but I get TinyMCE full features. Great ! with all the tags I am used to (P BR &nbsp marks etc. and creating tables.) Tested on Chrome, Edge and FireFox. French plugin installed : https://fr.wordpress.org/plugins/tinymce-advanced/

Do you have the latest WP version? I see comments in TinyMCE support section that some things break with latest WP.

I will give it a try as soon as they add an update.

Maybe notifying Elementor team about this very much required update would help us all as well :)

**Comment by ReyStub** [comment]
Yesterday I was on WP v5.8.3, but today as you suggested @deepou I updated to WP v5.9
and everything still works perfectly fine.
To summarize my configuration, find below the different versions, everything is up to date
WP: v5.9
Theme: Neve v3.1.4
Elementor (not Pro): v3.5.4
Advanced Editor Tools: v5.6.0
Server: Linux/Apache
PHP: v8.0.14

For notifying Elementor I will take a look at their forum soon..
Good luck :)

**Comment by coadr93** [comment]
Fix this Elementor! Also in templates, when you use import feature of post content or product content, the <p> tags don't show and everything is one big chunk of text.

**Comment by LogoLogics2** [comment]
I have the same issue in 2024! The fix with https://nl.wordpress.org/plugins/tinymce-advanced/ works fine!

"I have installed extension "Advanced Editor Tools" (previously TinyMCE Advanced), then in its settings activated "Preserve Paragraph Tags in Classic Block and Classic Editor" (this is English translation from my French environment) and it works perfectly. Not only it solves the problem, but I get TinyMCE full features. Great ! with all the tags I am used to (P BR &nbsp marks etc. and creating tables.) Tested on Chrome, Edge and FireFox."

**Comment by eyal1717** [comment]
I've installed and activated Advanced Editor Tools plugin, but not only that the problem still persist, it has worsen. All pages at my Physical therapy blog start changing without even being previously edited. It demands a constant cache purge. It's 2025 and indeed, it's about time someone in elementor will wake up.
Any ideas what to do?
